pkg/globalstats/stats.go

```diff
@@ -49,45 +53,81 @@ func Incr(name string, tags ...stats.Tag) {
 	engine.Incr(name, tags...)
 }
 
+func Disable() {
+	mut.Lock()
+	defer mut.Unlock()
+
+	if flusherStop != nil {
+		close(flusherStop)
+	}
+}
+
 func Initialize(ctx context.Context, config Config) {
-	once.Do(func() {
-		if config.AppName == "" {
+	mut.Lock()
+	defer mut.Unlock()
+
+	if config.AppName == "" {
+		if len(os.Args) > 0 {
+			config.AppName = filepath.Base(os.Args[0])
+		} else {
 			config.AppName = "unknown"
```

Comment:
I don't think `len(os.Args)` can be `0`, but CMIIW. Assuming better safe than sorry :)